src/Symfony/Security/State/LinkAccessCheckerProvider.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Symfony\Security\State;
+
+use ApiPlatform\Metadata\HttpOperation;
+use ApiPlatform\Metadata\Link;
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\State\ProviderInterface;
+use ApiPlatform\Symfony\Security\Exception\AccessDeniedException;
+use ApiPlatform\Symfony\Security\ResourceAccessCheckerInterface;
+
+class LinkAccessCheckerProvider implements ProviderInterface
```

Comment:
```suggestion

/**
 * class description
 * @experimental
 */
final class LinkAccessCheckerProvider implements ProviderInterface
```

I'd suggest to make these 2 provider experimental until 4.x, then we'll enable the option to `true`. This allows us to make modification until then if needed.